sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -457,17 +457,25 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
           break;
        """)
 
+    val switchCode = if (caseBranches.size > 0) {
```

Comment:
We don't need to compute `caseBranches`, too, if hset is empty?
```
    val valueGen = child.genCode(ctx)
    val switchCode = if (hset.isEmpty) {
      val caseValuesGen = hset.filter(_ != null).map(Literal(_).genCode(ctx))
      val caseBranches = caseValuesGen.map(literal =>
        code"""
          case ${literal.value}:
            ${ev.value} = true;
            break;
       """)

      code"""
        switch (${valueGen.value}) {
          ${caseBranches.mkString("\n")}
          default:
            ${ev.isNull} = $hasNull;
        }
       """
    } else {
      s"${ev.isNull} = $hasNull;"
    }
```

Reply:
btw, can we fold this expr in the optimizer when hset is empty?

Reply:
if `hset` is empty, the computation of `caseBranches` is noop, right?

Reply:
This is a special case. Usually `InSet` wouldn't have empty set, as we have a threshold to convert `In` to `Inset`.

Reply:
Ur, I see.